typescript/sdk/src/warp/WarpCore.ts

```diff
@@ -584,6 +584,23 @@ export class WarpCore {
       destinationBalance.toString(),
     );
 
+    // check for scaling factor
+    if (
+      originToken.scale &&
+      destinationToken.scale &&
+      originToken.scale !== destinationToken.scale
+    ) {
+      const precisionFactor = 100_000;
+      const scaledAmount =
+        amount *
+        BigInt((originToken.scale * precisionFactor) / destinationToken.scale);
+
+      return (
+        BigInt(Number(destinationBalanceInOriginDecimals) * precisionFactor) >=
```

Comment:
iiuc, `destinationBalanceInOriginDecimals` is a BigNumber. does that mean converting to a Number might overflow